test/e2e/helm_test.go

```diff
@@ -218,6 +220,154 @@ var _ = Describe("Workload cluster creation", func() {
+						HelmChartProxy:        hcp,
+					}
+				})
+			})
+
+			// Update existing Helm chart and expect Helm release to remain unchanged
+			By("Updating nginx HelmChartProxy valuesTemplate", func() {
+				hcp.Spec.ValuesTemplate = newNginxValues
+				HelmReleaseUnchangedSpec(ctx, func() HelmReleaseUnchangedInput {
+					return HelmReleaseUnchangedInput{
+						BootstrapClusterProxy: bootstrapClusterProxy,
+						Namespace:             namespace,
+						ClusterName:           clusterName,
+						HelmChartProxy:        hcp,
+					}
+				})
+			})
+
+			// Modify Helm release and expect it to remain unchanged
+			By("Upgrading Helm release out-of-band first time", func() {
+				HelmReleaseOutOfBandSpec(ctx, func() HelmReleaseOutOfBandInput {
+					return HelmReleaseOutOfBandInput{
+						BootstrapClusterProxy: bootstrapClusterProxy,
+						Namespace:             namespace,
+						ClusterName:           clusterName,
+						HelmChartProxy:        hcp,
+						Values:                []string{"--controller.name=new-nginx-controller-1"},
+						WaitPeriod:            installOnceWaitPeriod,
+					}
+				})
+			})
+
+			// Modify Helm release and expect it to remain unchanged
+			By("Upgrading Helm release out-of-band second time", func() {
+				HelmReleaseOutOfBandSpec(ctx, func() HelmReleaseOutOfBandInput {
+					return HelmReleaseOutOfBandInput{
+						BootstrapClusterProxy: bootstrapClusterProxy,
+						Namespace:             namespace,
+						ClusterName:           clusterName,
+						HelmChartProxy:        hcp,
+						Values:                []string{"--controller.name=new-nginx-controller-2"},
+						WaitPeriod:            installOnceWaitPeriod,
+					}
+				})
+			})
+
+			// Modify Helm release and expect it to remain unchanged
+			By("Upgrading Helm release out-of-band third time", func() {
+				HelmReleaseOutOfBandSpec(ctx, func() HelmReleaseOutOfBandInput {
+					return HelmReleaseOutOfBandInput{
+						BootstrapClusterProxy: bootstrapClusterProxy,
+						Namespace:             namespace,
+						ClusterName:           clusterName,
+						HelmChartProxy:        hcp,
+						Values:                []string{"--controller.name=new-nginx-controller-3"},
+						WaitPeriod:            installOnceWaitPeriod,
+					}
+				})
+			})
+
+			// Modify Helm release and expect it to remain unchanged
+			By("Upgrading Helm release out-of-band fourth time", func() {
+				HelmReleaseOutOfBandSpec(ctx, func() HelmReleaseOutOfBandInput {
+					return HelmReleaseOutOfBandInput{
+						BootstrapClusterProxy: bootstrapClusterProxy,
+						Namespace:             namespace,
+						ClusterName:           clusterName,
+						HelmChartProxy:        hcp,
+						Values:                []string{"--controller.name=new-nginx-controller-4"},
+						WaitPeriod:            installOnceWaitPeriod,
+					}
+				})
+			})
+
+			// Modify Helm release and expect it to remain unchanged
+			By("Upgrading Helm release out-of-band fifth time", func() {
```

Comment:
Any particular reason why we upgraded the Helm release five times (instead of just once)? Maybe we could make a loop out of these five `// Modify Helm release` sections.

Reply:
I just wanted to be sure that subsequent changes to the Helm chart would still be durable. I think it makes sense to put it in a loop, though.